packages/dataviews/src/dataviews-layouts/picker-table/index.tsx

```diff
@@ -0,0 +1,445 @@
+import DataViewsSelectionCheckbox from '../../components/dataviews-selection-checkbox';
+import { useIsMultiselectPicker } from '../../components/dataviews-picker/footer';
+import { sortValues } from '../../constants';
+import type {
+	NormalizedField,
+	ViewPickerTable as ViewPickerTableType,
+	ViewPickerTableProps,
+} from '../../types';
+import type { SetSelection } from '../../types/private';
+import ColumnHeaderMenu from '../table/column-header-menu';
+import ColumnPrimary from '../table/column-primary';
+import getDataByGroup from '../utils/get-data-by-group';
+
+interface TableColumnFieldProps< Item > {
+	fields: NormalizedField< Item >[];
+	column: string;
+	item: Item;
+	align?: 'start' | 'center' | 'end';
+}
+
+interface TableRowProps< Item > {
+	item: Item;
+	fields: NormalizedField< Item >[];
+	id: string;
+	view: ViewPickerTableType;
+	titleField?: NormalizedField< Item >;
+	mediaField?: NormalizedField< Item >;
+	descriptionField?: NormalizedField< Item >;
+	selection: string[];
+	getItemId: ( item: Item ) => string;
+	onChangeSelection: SetSelection;
+	multiselect: boolean;
+	posinset?: number;
+}
+
+function TableColumnField< Item >( {
+	item,
+	fields,
+	column,
+	align,
+}: TableColumnFieldProps< Item > ) {
+	const field = fields.find( ( f ) => f.id === column );
+
+	if ( ! field ) {
+		return null;
+	}
+
+	const className = clsx( 'dataviews-view-table__cell-content-wrapper', {
+		'dataviews-view-table__cell-align-end': align === 'end',
+		'dataviews-view-table__cell-align-center': align === 'center',
+	} );
+
+	return (
+		<div className={ className }>
+			<field.render item={ item } field={ field } />
+		</div>
+	);
+}
+
+function TableRow< Item >( {
+	item,
+	fields,
+	id,
+	view,
+	titleField,
+	mediaField,
+	descriptionField,
+	selection,
+	getItemId,
+	onChangeSelection,
+	multiselect,
+	posinset,
+}: TableRowProps< Item > ) {
+	const { paginationInfo } = useContext( DataViewsContext );
+	const isSelected = selection.includes( id );
+	const [ isHovered, setIsHovered ] = useState( false );
+	const {
+		showTitle = true,
+		showMedia = true,
+		showDescription = true,
+		infiniteScrollEnabled,
+	} = view;
+	const handleMouseEnter = () => {
+		setIsHovered( true );
+	};
+	const handleMouseLeave = () => {
+		setIsHovered( false );
+	};
+
+	const columns = view.fields ?? [];
+	const hasPrimaryColumn =
+		( titleField && showTitle ) ||
+		( mediaField && showMedia ) ||
+		( descriptionField && showDescription );
+
+	return (
+		<tr
+			className={ clsx( 'dataviews-view-table__row', {
+				'is-selected': isSelected,
+				'is-hovered': isHovered,
+			} ) }
+			onMouseEnter={ handleMouseEnter }
+			onMouseLeave={ handleMouseLeave }
+			aria-setsize={
+				infiniteScrollEnabled ? paginationInfo.totalItems : undefined
+			}
+			aria-posinset={ posinset }
+			role={ infiniteScrollEnabled ? 'article' : undefined }
+			onClick={ ( event ) => {
+				if ( event.target instanceof HTMLElement ) {
```

Comment:
I think this will be `true` most of the time.

What if it checks if the click came from the checkbox (or other interactive elements)?